### DISCUSSION TITLE
Have CHARGE later than time == 0:00 ?

### DISCUSSION THREAD

**Discussion by einarbjorkman** [artisan-roaster-scope/artisan#968] [Q&A] [answered]
Hello,
I have created a few custom roast profiles which I use for SV's for a PID in background mode.
However when I press START Artisan starts logging the data from the CHARGE point (so CHARGE = 0:00 time).
I have tried both Auto Charge and clicking charge fast, however Auto Charge seems to not work immediately (I've also read that it needs about 15 seconds of recording before CHARGE) and I am too slow with pressing - the result being that my SV's are staggered to the right of the background roast profile.
I have tried shuffling the background roast profile to the right by one minute with the timing tool in Roast >> Background but as soon as I hit START the program seems to reset to make CHARGE the 0:00 time mark.
Is there a way to allow me to press START, have the time be 0:00 (or something negative?) and then have the CHARGE event at a later time such as 01:00?
Help would be much appreciated!
/ Einar

**Comment by MAKOMO** [comment]
Not sure I understand your issue, but Artisan starts logging at 00:00 always on START. Once the CHARGE event is set (manually or automatically), this defines the "new" time point 00:00 and the begin of the recording gets assigned the corresponding negative time. Thus for Artisan the time point 00:00 is always at the CHARGE point (begin of the actual roast), but if no such point is defined this point correspond to the begin of the recording (which usually is before the actual start of the roast).

By default the background roast is aligned to the CHARGE event once it is set for the foreground roast. Up to this point the background profile is not aligned in any way. There is an option to influence this alignment in the Background dialog (menu Config >> Background).

Well, that all sounds confusing as well, but I hope it helps a bit to clarify Artisans behavior.

**Comment by einarbjorkman** [comment]
Hi thank you for your fast response!
Let me see if I can clarify the issue a bit:
![image](https://user-images.githubusercontent.com/104577724/192277068-b79b572b-f981-40ff-a9de-a1e7a1885cfe.png)
This screenshot was taken a few seconds after I hit START.
As you can see when I clicked START the recording started right at the CHARGE event.
I would like to be able to hit START and then for the CHARGE event to come 1 minute later.
![image](https://user-images.githubusercontent.com/104577724/192277456-8e6e70bd-74aa-4251-8d90-ced99763dc90.png)
Like the picture above...
I hope that make sense and explains what I mean (?),

Cheers / Einar

**Comment by MAKOMO** [comment]
What about adding an alarm that triggers CHARGE 1min after START?

<img width="1035" alt="Screenshot 2022-09-26 at 15 08 52" src="https://user-images.githubusercontent.com/13909011/192284860-bed8ab9b-3e60-4989-9aaa-d09d96d08ba9.png">

**Reply by einarbjorkman** [reply]
Hello and thank you for the suggestion.
I have tried replicating the settings you shared for an alarm however this did not have the desired effect for me (timer still starts at the CHARGE event),
Cheers / Einar

**Reply by MAKOMO** [reply]
I am again confused by your answer. Btw. which version of Artisan are you using?

As said before, the timer starts once START is pressed. Once CHARGE is set (in this case by the alarm) that time point defines the 00:00. 

Do you want to have the CHARGE event to be located at 01:00!?
Can you explain why?

**Reply by einarbjorkman** [reply]
I am using version 2.6.0

"Do you want to have the CHARGE event to be located at 01:00!?"
Yes this is what I would want if possible.
The reason is that now:
1) When I press START, the program starts recording right at my background curve.
2) Then CHARGE is not activated fast enough so that the SV curve is a bit different from the background curve.
3) Auto Charge does not solve this consistently
4) So I would like to have the "time space" to press CHARGE manually
Does that make sense?

Regards / Einar

**Reply by einarbjorkman** [reply]
Actually I downloaded a roast profile from the internet where this happens automatically.
So I am trying to make my custom profile do the same thing...

**Comment by einarbjorkman** [comment]
![image](https://user-images.githubusercontent.com/104577724/192294535-2b7504a6-6535-4c37-aff1-3b59aa971917.png)
This is how I would like it to happen :)

**Reply by MAKOMO** [reply]
I still don't get it. If you press START, you have any time to press CHARGE (which you should do at the begin of the roast). At that very moment. Artisan will re-align the background curve such that the backgrounds CHARGE event is aligned at 00:00 with the CHARGE event of the profile you are currently roasting. This is what you want, right? Everything else does not make much sense, or does it?

**Reply by einarbjorkman** [reply]
You can mark this as answer ;)

**Comment by einarbjorkman** [comment, marked as answer]
Oh I get it what your trying to tell me now!
Sorry for being so thick 😅
Thank you for the detailed explanation and for your time :)
Cheers / Einar
